File: Add Code Here/PYTHON/Empty_Cell_in_a_Matrix.py
def emptyCells(N, K, tasks):

	matrix = [[-1 for i in range(N)]for j in range(N)]
	output = []

	for _ in tasks:
		x,y = _ 

		for i in range(N):
			matrix[x][i] = 0

		for i in range(N):
			matrix[i][y] = 0

		count = 0
		for arr in matrix:
			for i in arr:
				if i == -1:
					count += 1
		output.append(count)

	return output

test_cases = int(input())

while test_cases:

	N, K = map(int, input().split())
	tasks = []

	for t in range(K):
		i,j = map(int,input().split())
		X = i,j
		tasks.append(X)

	output = emptyCells(N, K, tasks)

	for emptyCell in output:
		print(emptyCell, end=' ')

	print()
	test_cases -= 1

File: Add Code Here/C++/Empty_cell_in_a_Matrix.cpp
vector<long> emptyCells(int n, int k, vector<vector<int>> &tasks)
{
	vector<vector<int>> matrix(n, vector<int>(n, -1));

	vector<long> ans(k);

	for (int i = 0 ; i < k ; i++)
	{
		int x = tasks[i][0];
		int y = tasks[i][1];

		for (int j = 0 ; j < n ; j++)
		{
			matrix[x][j] = 0;
			matrix[j][y] = 0;
		}

		long count = 0;
		for (int j = 0; j < n; j++)
		{
			for (int p = 0; p < n; p++)
			{
				if (matrix[j][p] == -1)
				{
					count++;
				}
			}
		}

		ans[i] = count;
	}
	return ans;
}